pkgs/development/python-modules/python-dbusmock/default.nix

```diff
@@ -53,11 +54,13 @@ buildPythonPackage rec {
 
   checkPhase = ''
     runHook preCheck
-    export PATH="$PATH:${lib.getOutput "test" bluez}/test";
+    ${if doCheck then ''export PATH="$PATH:${lib.getOutput "test" bluez}/test'' else ""}";
```

Comment:
Something like the following (untested) will be even cleaner IMO:

```patch
diff --git a/pkgs/development/libraries/polkit/default.nix b/pkgs/development/libraries/polkit/default.nix
index e9e3d94327b..9e600ee9f19 100644
--- a/pkgs/development/libraries/polkit/default.nix
+++ b/pkgs/development/libraries/polkit/default.nix
@@ -99,8 +99,10 @@ stdenv.mkDerivation rec {
     rsync
     (python3.withPackages (pp: with pp; [
       dbus-python
-      # Avoid dependency cycle.
-      (python-dbusmock.override { doCheck = false; })
+      (python-dbusmock.overridePythonAttrs {
+        # Avoid dependency cycle.
+        doCheck = false;
+      })
     ]))
 
     # man pages
diff --git a/pkgs/development/python-modules/python-dbusmock/default.nix b/pkgs/development/python-modules/python-dbusmock/default.nix
index ae5db2aebfe..871c25c4909 100644
--- a/pkgs/development/python-modules/python-dbusmock/default.nix
+++ b/pkgs/development/python-modules/python-dbusmock/default.nix
@@ -1,16 +1,22 @@
 { lib, buildPythonPackage, fetchFromGitHub, runtimeShell,
   nose, dbus, dbus-python, pygobject3,
   which, pyflakes, pycodestyle, bluez, networkmanager,
-  doCheck ? true
+  runCommand
 }:
 
+let
+  pbap-client = runCommand "pbap-client" { } ''
+    mkdir -p "$out/bin"
+    ln -s "${bluez.test}/test/pbap-client" "$out/bin/pbap-client"
+  '';
+in
 buildPythonPackage rec {
   pname = "python-dbusmock";
   version = "0.26.1";
 
   src = fetchFromGitHub {
     owner = "martinpitt";
-    repo = pname;
+    repo = "python-dbusmock";
     rev = version;
     sha256 = "sha256-kavbWMTgKU/rBIo7RMs9NkwReYQyEdeFwMBSzEM9wa0=";
   };
@@ -49,18 +55,17 @@ buildPythonPackage rec {
 
   checkInputs = [
     nose dbus dbus-python which pycodestyle pyflakes
-    pygobject3 bluez (lib.getOutput "test" bluez) networkmanager
+    pygobject3 bluez
+    pbap-client
+    networkmanager
   ];
 
   checkPhase = ''
     runHook preCheck
-    ${if doCheck then ''export PATH="$PATH:${lib.getOutput "test" bluez}/test'' else ""}";
     nosetests -v
     runHook postCheck
   '';
 
-  inherit doCheck;
-
   meta = with lib; {
     description = "Mock D-Bus objects for tests";
     homepage = "https://github.com/martinpitt/python-dbusmock";
```

Reply:
Looks good, this builds for me!

Reply:
ofborg is unhappy with this, should I use the original test attribute acquisition method? 
`lib.getOutput "test" bluez`

Reply:
That would fix evaluation by just falling back to `out` output, which does not contain the file.

I guess the issue is that we are limiting the test output to `x86_64`:

https://github.com/NixOS/nixpkgs/blob/025f00750d07df498b42bd6ebea38c6802fe9857/pkgs/os-specific/linux/bluez/default.nix#L108

Removing the output was introduced in:

https://github.com/NixOS/nixpkgs/commit/0bbdba2d119caa0fccf48529cf374df7075403d2

but it does not make much sense to me.

I would probably make installing the `test` output unconditional.